### PR TITLE
Search

### PR DESCRIPTION
**Pull request by manst**
@squallstar 
Users should be able to search all files they have access to, see a list and the corresponding folder
ref [Fliplet/fliplet-studio#864](https://github.com/Fliplet/fliplet-studio/issues/864)

### CODE REVIEW

File: js/interface.js [hunk 1]
```diff
@@ -555,6 +577,188 @@ function uploadFiles(files) {
...
+    showNothingFoundAlert(true);
+    return;
+  }
+
+  if (searchType == 'all') {
+    resetToTop();
+  }
+
+  result = result
+      .filter(function (item) {
```
Comment (squallstar) [inline js/interface.js]: indentation is two spaces, not four

Comment (manst) [reply js/interface.js]: Fixed

File: css/interface.css [hunk 2]
```diff
@@ -696,6 +696,21 @@
   text-align: center;
 }
 
+.file-path{
```
Comment (squallstar) [inline css/interface.css]: space before the curly bracker

File: interface.html [hunk 3]
```diff
@@ -20,6 +20,24 @@
...
           </div>
         </div>
+        <div class="search-holder">
+          <select class="form-control search-type">
+            <option value="all">All</option>
+            <option value="this-folder" selected>This folder</option>
+          </select>
+          <input type="text" class="form-control search-term" placeholder="Search...">
+        </div>
+        <!--<div class="search form-inline">
```
Comment (squallstar) [inline interface.html]: what's this comment for?

File: interface.html [hunk 4]
```diff
@@ -95,7 +113,17 @@
                       </div>
                     </div>
                     -->
-                    <div class="file-cell">
+                    <div class="file-cell search-result-cell">
+                      <div class="file-path">
+                        Path
```
Comment (squallstar) [inline interface.html]: what does `Path` display? can you post a screenshot?

File: js/interface.js [hunk 5]
```diff
@@ -513,6 +529,12 @@ function resetUpTo(element) {
   updatePaths();
 }
 
+//Resets navigation to first item
```
Comment (squallstar) [inline js/interface.js]: space after `//`

File: js/interface.js [hunk 6]
```diff
@@ -838,6 +1036,52 @@ $('.file-manager-wrapper')
...
+      disableSearchState();
+      backToLastFolderBeforeSearch();
+      return;
+    }
+
+    removeSelection();
+    hideSideActions();
+
+    var type = $searchType.val();
+    search(type, term);
```
Comment (squallstar) [inline js/interface.js]: doesn't look like `.catch` is handled with an error

File: interface.html (hunk 4 above)
Comment (manst) [reply interface.html]: image

File: css/interface.css (hunk 2 above)
Comment (manst) [reply css/interface.css]: Fixed

File: interface.html (hunk 3 above)
Comment (manst) [reply interface.html]: It was a temp implementation, removed.

File: js/interface.js (hunk 5 above)
Comment (manst) [reply js/interface.js]: Fixed

File: js/interface.js (hunk 6 above)
Comment (manst) [reply js/interface.js]: Added catch block

**Pull request merged**